src/core/components/ping-pull-stream.js

```diff
@@ -27,7 +27,7 @@ module.exports = function pingPullStream (self) {
       if (err) {
         log.error(err)
         source.push(getPacket({ success: false, text: err.toString() }))
-        source.end(err)
+        source.end()
```

Comment:
The errors are part of the API. passing the error was making it so that the response would never get to the client!